pkg/eks/eks.go

```diff
@@ -274,7 +274,16 @@ func (c *ClusterProvider) NewOpenIDConnectManager(spec *api.ClusterConfig) (*iam
 		return nil, fmt.Errorf("unknown EKS ARN: %q", spec.Status.ARN)
 	}
 
-	return iamoidc.NewOpenIDConnectManager(c.Provider.IAM(), parsedARN.AccountID, *c.Status.ClusterInfo.Cluster.Identity.Oidc.Issuer, parsedARN.Partition)
+	return iamoidc.NewOpenIDConnectManager(c.Provider.IAM(), parsedARN.AccountID,
+		*c.Status.ClusterInfo.Cluster.Identity.Oidc.Issuer, parsedARN.Partition, sharedTags(c.Status.ClusterInfo.Cluster))
+}
+
+func sharedTags(cluster *awseks.Cluster) map[string]string {
+	return map[string]string{
+		api.ClusterNameTag:   *cluster.Name,
+		api.EksctlVersionTag: version.GetVersion(),
+	}
```

Comment:
since users are not providing these, can we just set an array of `*awsiam.Tag` here and pass it through, rather than creating a `map[string]string` and then looping to translate in `pkg/iam/oidc/api.go`?

Reply:
I wouldn't be too keen to do that, `tags` are best represented as a `map[string]string` rather than a `[]*awsiam.Tag`, and that `Tag` type is specifically made for the AWS IAM APIs. It also breaks the abstraction.